examples/colormap_normalization/colormap_normalizations_piecewisenorm.py

```diff
@@ -0,0 +1,103 @@
+"""
+============================================
+Examples of arbitrary colormap normalization
+============================================
+
+Here I plot an image array with data spanning for a large dynamic range,
+using different normalizations. Look at how each of them enhances
+different features.
+
+"""
+
+
+import matplotlib.cm as cm
+import matplotlib.colors as colors
+import matplotlib.pyplot as plt
+
+from mpl_toolkits.mplot3d import Axes3D
+
+import numpy as np
+
+from sampledata import PiecewiseNormData
+
+X, Y, data = PiecewiseNormData()
+cmap = cm.spectral
+
+# Creating functions for plotting
+
+
+def make_plot(norm, label=''):
```

Comment:
>while the equivalent npy file is 1 MB, and even if I reduce the resolution a bit , or the variable type, it will not get smaller than 100 KB

What about if you zip it down? (And I think you can have a 100x100 array or something)

Can you address #2 in the docstring/a short comment? I think #3 is addressed with the linear normalization vs. fancy normalization plot.